### PR TITLE
fix(carthage): shell login

### PR DESCRIPTION
**Pull request by bimawa**
Force login shell. Looks we need it for Apple Silicon CPU. 

#1167

### CODE REVIEW

**Comment by yonaskolb** [comment]
Do you have more details on why this is required? Also before accepting this PR needs a changelog entry

**Comment by bimawa** [comment]
I don't know. But last xCode version not throw environments to Shell scripts. Google say so for that feature we should init /bin/sh with -l Login parameter. And after that, Xcode will throw OS environments to builds sh scripts.
Without this parameters $PATH not visible while sh scripts run. And we got errors, "carthage command not found" etc.
But this issue not reproduced on old Intel macOS. Maybe it's related with Apple CPU`s. 

Also, maybe we should make this property possible to modify? But `/bin/sh with -l ` also works good on intel mac's in my team.

**Comment by bimawa** [comment]
@yonaskolb  Did you know why fails testes? 
`CI / Xcode 12.5.1 (pull_request) Failing after 2m — Xcode 12.5.1`
This xcode 12.5.1 didn't support login shell?

**Comment by yonaskolb** [comment]
The fixture snapshot tests are failing because the diffs haven't been committed. Run the tests locally and push the diffs

**Review by yonaskolb — COMMENTED** [review]
Thanks. Could you also add a changelog entry under 'Next version'

**Comment by bimawa** [comment]
@yonaskolb like that?

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@
 - Ensure fileTypes are mapped to JSON value [#1112](https://github.com/yonaskolb/XcodeGen/pull/1112) @namolnad
 - Fix platform filter for package dependecies [#1123](https://github.com/yonaskolb/XcodeGen/pull/1123) @raptorxcz
 - Fix Xcode 13 build [#1130](https://github.com/yonaskolb/XcodeGen/issues/1127) @raptorxcz @mthole
+- Fix Monterey MacOS shell version, shell login flag for environments [#1167](https://github.com/yonaskolb/XcodeGen/issues/1167) @bimawa
```
Comment (yonaskolb) [inline CHANGELOG.md:24]: Wrong version, but I'll move it

**Review by yonaskolb — APPROVED** [review]

**Pull request merged**